File: study-ai-myapp-fimaly/frontend/android/lib/models/family.dart
class Family {
  final int id;
  final String name;
  final String? description;
  final String? avatar;
  final int creatorId;
  final DateTime createdAt;
  final DateTime? updatedAt;

  Family({
    required this.id,
    required this.name,
    this.description,
    this.avatar,
    required this.creatorId,
    required this.createdAt,
    this.updatedAt,
  });

  factory Family.fromJson(Map<String, dynamic> json) {
    return Family(
      id: json['id'],
      name: json['name'],
      description: json['description'],
      avatar: json['avatar'],
      creatorId: json['creatorId'],
      createdAt: DateTime.parse(json['createdAt']),
      updatedAt: json['updatedAt'] != null ? DateTime.parse(json['updatedAt']) : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'description': description,
      'avatar': avatar,
      'creatorId': creatorId,
      'createdAt': createdAt.toIso8601String(),
      'updatedAt': updatedAt?.toIso8601String(),
    };
  }
}

File: study-ai-best-practice-dev-flow/best-practice-dev-flow-backend/src/main/java/com/bage/study/ai/best/practice/dev/flow/entity/Family.java
package com.bage.study.ai.best.practice.dev.flow.entity;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;

import java.time.LocalDateTime;

@Entity
@Table(name = "sys_family")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Family {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 100)
    private String name;

    @Column(length = 500)
    private String description;

    @Column(length = 512)
    private String avatar;

    @Column(name = "creator_id", nullable = false)
    private Long creatorId;

    @Column(name = "administrator_id")
    private Long administratorId;

    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @Column(name = "updated_at", nullable = false)
    private LocalDateTime updatedAt;

    @Column(name = "created_by", length = 64)
    private String createdBy;

    @Column(name = "updated_by", length = 64)
    private String updatedBy;

    @Column(columnDefinition = "TINYINT DEFAULT 0")
    private Integer deleted = 0;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
        if (administratorId == null) {
            administratorId = creatorId;
        }
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}